java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -522,7 +540,137 @@ public static ColumnVector stringConcatenate(Scalar separator, Scalar narep, Col
       column_views[i] = columns[i].getNativeView();
     }
 
-    return new ColumnVector(stringConcatenation(column_views, separator.getScalarHandle(), narep.getScalarHandle()));
+    return new ColumnVector(stringConcatenation(column_views, separator.getScalarHandle(),
+        narep.getScalarHandle(), separate_nulls));
+  }
+
+  /**
+   * Concatenate columns of strings together using a separator specified for each row
+   * and returns the result as a string column. If the row separator for a given row is null,
+   * output column for that row is null. Null column values for a given row are skipped.
+   * @param columns array of columns containing strings, must be more than 1 columns
+   * @param sep_col strings column that provides the separator for a given row
+   * @return A new java column vector containing the concatenated strings with separator between.
+   */
+  public static ColumnVector stringConcatenate(ColumnView[] columns, ColumnView sep_col) {
+    try (Scalar nullString = Scalar.fromString(null);
+         Scalar emptyString = Scalar.fromString("")) {
+      return stringConcatenate(columns, sep_col, nullString, emptyString, false);
+    }
+  }
+
+  /**
+   * Concatenate columns of strings together using a separator specified for each row
+   * and returns the result as a string column. If the row separator for a given row is null,
+   * output column for that row is null unless separator_narep is provided.
+   * The separator is applied between two output row values if the separate_nulls
+   * is `YES` or only between valid rows if separate_nulls is `NO`.
+   * @param columns array of columns containing strings, must be more than 1 columns
+   * @param sep_col strings column that provides the separator for a given row
+   * @param separator_narep String that should be used in place of a null separator for a given
+   *                        row. 
+   * @param col_narep string that should be used in place of any null strings
+   *                  found in any column.
+   * @param separate_nulls if true, then the separator is included for null rows if
+   *                       `col_narep` is valid.
+   * @return A new java column vector containing the concatenated strings with separator between.
+   */
+  public static ColumnVector stringConcatenate(ColumnView[] columns,
+      ColumnView sep_col, Scalar separator_narep, Scalar col_narep, boolean separate_nulls) {
+    assert columns.length >= 1 : ".stringConcatenate() operation requires at least 1 column";
```

Comment:
Docs say must be more than one column but this allows 1

Reply:
thanks for catching, need to update docs

Reply:
concat also supports one or more now, so that was a really old issue. We should update the concat docs too.

Reply:
it looks like that was updated yesterday (https://github.com/rapidsai/cudf/commit/7af8966e705b1da5d5c694e063d89a47b88e3b9c)  and even though I copied it, I updated it, then later changed when I realized spark supported